### PR TITLE
python3Packages.mcstatus: 5.2.0 -> 6.0.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 6.0.0

Change log: https://github.com/Dinnerbone/mcstatus/releases/tag/v6.0.0

- Related Home Assistant change: https://github.com/home-assistant/core/pull/51517
- Target Home Assistant release: current (only dependency update, no code changes)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125644` at 637c61c0 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/4f0191d4706dd8e50fc33490ef909f77">home-assistant</a></li>
    <li><a href="https://gist.github.com/d3632837aec2da1f4f03c5218fd66838">python38Packages.aioguardian</a></li>
  </ul>
</details>
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>python38Packages.asyncio-dgram</li>
    <li>python38Packages.mcstatus</li>
    <li>python38Packages.pywizlight</li>
    <li>python39Packages.asyncio-dgram</li>
    <li>python39Packages.mcstatus</li>
    <li>python39Packages.pywizlight</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'dnspython3' doesn't match anything in file 'requirements.txt'
Near pkgs/development/python-modules/mcstatus/default.nix:46:0:
```
   |
46 |     description = "Python library for checking the status of Minecraft servers";
   | ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 125644` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python38Packages.aioguardian</li>
    <li>python38Packages.asyncio-dgram</li>
    <li>python38Packages.mcstatus</li>
    <li>python38Packages.pywizlight</li>
    <li>python39Packages.asyncio-dgram</li>
    <li>python39Packages.mcstatus</li>
    <li>python39Packages.pywizlight</li>
  </ul>
</details>

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15, nixos x86_64

File: pkgs/development/python-modules/aioguardian/default.nix [hunk 1]
```diff
@@ -27,7 +27,9 @@ buildPythonPackage rec {
 
   format = "pyproject";
 
-  nativeBuildInputs = [ poetry ];
+  nativeBuildInputs = [
+    poetry
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/aioguardian/default.nix:31]: Not poetry-core?

Comment (fabaff) [reply pkgs/development/python-modules/aioguardian/default.nix:31]: Not at the moment. The build backend was changed already but the list of requirements is not up-to-date.

https://github.com/bachya/aioguardian/pull/67

**Pull request merged**